src/mpid/ch4/netmod/ofi/ofi_rma.h

```diff
@@ -292,6 +292,21 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_do_put(const void *origin_addr,
         goto fn_exit;
     }
 
+    /* The user should not force NATIVE if the program will issue sparse noncontig data.
+     * Thus, we simply use the nopack routine. To gain performance, the user should not force NATIVE. */
+    if (MPIDI_OFI_WIN(win).force_native_flag) {
```

Comment:
Suggest better comment:
```
/* If user force NATIVE then we can't fallback to AM. Since here we are dealing with low density target datatype,
 * and potentially low density origin datatype, the performance may be bad.
 */
```
Comment inside the code can't serve as "user suggestion" :) .

Reply:
There is no way to negotiate with user at runtime:-) Thus, I put the warning in log.